web/templates/article

```diff
@@ -206,7 +206,7 @@ in the Leanpub book store.
 {{- $is_blog := (eq .Group "blog") -}}
 
 {{- $not_index_page := (ne .FilenameWithoutExt "101") -}}
-{{- $is_book_page := (or (lt (len .FilenameWithoutExt) 4) (ne (slice .FilenameWithoutExt 0 4) "100-")) -}}
+{{- $is_book_page := (or (lt (len .FilenameWithoutExt) 4) (ne (slice .FilenameWithoutExt 0 3) "100-")) -}}
```

Comment:
Thanks for PR. But isn't "100-" a string with length 4?

Reply:
The docs says:

```
slice
	returns the result of slicing its first argument by the
	remaining arguments. Thus "slice x 1 2" is, in Go syntax, x[1:2],
	while "slice x" is x[:], "slice x 1" is x[1:], and "slice x 1 2 3"
	is x[1:2:3]. The first argument must be a string, slice, or array.
```

So, it is "length"?

Reply:
Could you elaborate a bit?
I thought you intended that slicing 4 chars from 0 to 3(length is 4) with `FilenameWithoutExt`.

Reply:
Oh, I got it.
I met some issue when I try to generate static files with `go run . -gen` with `101.html`  file 😃

Reply:
What is the issue?

Trace? Do you mean logging? You can use `{{ .Value }}` in the `article` template to show some values in the html output.

Reply:
Yes, below error happens when I try to generate static pages

```console
template: article:210:63: executing "promotions" at : error calling slice: index out of range: 4
```

I will check the reason of that again and then update this PR :smiley:

Reply:
I cloned https://github.com/go101korea/go101, and run `go run . -gen`, but it looks there is no such problem.

Reply:
I run it with arg as either 3 or 4.

Reply:
As you say, static generation in local works normally but it happens when I try to use it in GitHub Pages with publishing by GitHub Actions.

you can look this error with https://go101.or.kr/article/101.html

Reply:
That looks a bug before Go 1.18. Before Go 1.18, all arguments of `or` are evaluated whatever.
Since Go 1.18, if the an earlier argument is evaluated as `true`, then the following ones will not get evaluated.

Reply:
Yes, so I used it with Go 1.20 but it happens.

https://github.com/go101korea/go101/actions/runs/4123011097/jobs/7120507900

Reply:
Do you mean the problem still exists with Go 1.20?

Reply:
Yes, This error is happend with Go 1.20 as you can see in https://github.com/go101korea/go101/actions/runs/4123011097/jobs/7120507900

Reply:
If the generated pages are deployed to https://go101.or.kr properly, then it is weird.

Reply:
Have you checked the generated 101.html files in the ./generated folder after the build action ended?

Reply:
Yes, you can find it below link.

https://github.com/go101korea/go101/blob/public/article/101.html

Reply:
How do you confirm the used toolchain is 1.20?

Reply:
I specified 1.20 for GitHub Actions and you can confirm it with below link.

https://github.com/go101korea/go101/actions/runs/4123011097/jobs/7120507900

Reply:
Yes, That is really weird.

I rebased a new commit and published again.

- GitHub Actions: https://github.com/go101korea/go101/actions/runs/4141525657

and I just found something very strange in `go-version` meta HTML tag.

It should be 1.20 because I specified that in [static.yml](https://github.com/go101korea/go101/blob/master/.github/workflows/static.yml) and used Go 1.20 on building static sites for real as far as we shown below screen capture.

![image](https://user-images.githubusercontent.com/4374338/218024453-ae61a5b8-c9fc-49a6-8f50-79fd4952da65.png)

but generated sites showns `go-version` meta tag shows `1.17.13`.

![image](https://user-images.githubusercontent.com/4374338/218024520-0b6f0410-1ba6-4a31-972e-5d29b7983d5e.png)

so I'm going to find the reason of that!

Reply:
Hello!

I've find the reason of that. below line for overriding `PATH` envvar is that.

```yaml
      - name: Build site
        run: |
          ...
          PATH="$GOPATH/bin:$PATH" go run . -ge
```

['setup-go' GitHub Actions](https://github.com/actions/setup-go) installs the specified version of Go toolchain under `/opt/hostedtoolcache/go/<VERSION>/<ARCH>` directory but I had overrided `PATH` envvar with default `GOPATH` envvar.

After I removed it and try to generate again, it works gracefully :smiley: 

Thanks for your help!

Reply:
Glad to hear the problem is resolved. :)

I will close the PR soon.